### DISCUSSION TITLE
UDIM Visualisation - 3D  Viewport

### DISCUSSION THREAD

**Discussion by Kieruss** [StjerneIdioten/I3D-Blender-Addon#124] [Ideas & Feature Requests]
Problem: Can't see the colours if no diffuse map used

Current solution: Manually change the colours to somewhat match how it will look in-game

Great solution: (if possible) Set the colour and texture based on the UDIM mapping

**Comment by ssnd292** [comment]
That was already tried by a friend, but its not possible because Eevee currently doesnt support the amount of UDIM Tiles needed to display FS' UDIM Grid.

**Reply by kbrandwijk** [reply]
Wasn't that addressed in 2.82 already? https://developer.blender.org/T73133

**Reply by Kieruss** [reply]
Apparently so, since the box test is no longer pink :)
![image](https://user-images.githubusercontent.com/24835155/111720593-5239ad80-88b2-11eb-8a8a-e8d961a10b6c.png)

**Reply by StjerneIdioten** [reply]
> Wasn't that addressed in 2.82 already? https://developer.blender.org/T73133

Could be, the last time we talked about it was probably back in the 2.80 days.
Unless some testing has been going on at a later point that I wasn't aware of 😛 

> Apparently so, since the box test is no longer pink :)

Interesting 😉 
We actually discussed (half a year ago) making a sort of node-setup that would "approximate" the GE UDIM-setup, so that you could at least have an idea about it. Maybe that idea should make a comeback 😉

**Reply by kbrandwijk** [reply]
I wouldn't mind spending some time on making that happen. I've already experimented locally with shader setups to match the game engine, so integrating them into this addon would be kind of a natural transition.

**Reply by StjerneIdioten** [reply]
> I wouldn't mind spending some time on making that happen. I've already experimented locally with shader setups to match the game engine, so integrating them into this addon would be kind of a natural transition.

It just needs to be easily discernible from the stuff that we want to actually export, like the specular, normal and "glossmap" nodes. Currently it is all based on a name lookup of the BSDF node and that "glossmap" node standard I have defined.

Also any UI code should be a fairly safe bet to implement, I currently don't have any plans for changing much in that part of the codebase (with respect to the current rework I am doing)
